### PR TITLE
Client Side support for SAML Artifact Binding

### PR DESCRIPTION
**Pull request by IndunilRathnayake**
### Proposed changes in this pull request
Client side support for SAML artifact binding - Adding IDP UI configurations related to SAML artifact binding 

#### Documentation

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/util/IdentityApplicationConstants.java [hunk 1]
```diff
@@ -136,7 +136,7 @@ private OpenID() {
 
             public static final String UNSPECIFIED_NAME_ID_FORMAT =
                     "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified";
-
+            public static final String ARTIFACT_RESOLVE_URL="ArtifactResolveUrl";
```
Comment (lahirus) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/util/IdentityApplicationConstants.java]: minor formatting issue

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java [hunk 2]
```diff
@@ -1617,6 +1617,12 @@ private static void buildSAMLAuthenticationConfiguration(IdentityProvider fedIdp
         }
         properties.add(property);
 
+        property = new Property();
+        property.setName(IdentityApplicationConstants.Authenticator.SAML2SSO.ARTIFACT_RESOLVE_URL);
+        property.setValue(paramMap
```
Comment (lahirus) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java]: Looks like both lines can fit into one. Please check.

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp [hunk 3]
```diff
@@ -560,6 +564,24 @@
                         attributeConsumingServiceIndex = attributeConsumingServiceIndexProp.getValue();
                     }
 
+                    Property artifactBindingEnableProp = IdPManagementUIUtil.getProperty(fedAuthnConfig.getProperties(),
+                        IdentityApplicationConstants.Authenticator.SAML2SSO.IS_ARTIFACT_BINDING_ENABLED);
+                    if (artifactBindingEnableProp !=null) {
```
Comment (thanujalk) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp]: Format issue

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp [hunk 4]
```diff
@@ -560,6 +564,24 @@
...
+                        isArtifactBindingEnabled = Boolean.parseBoolean(artifactBindingEnableProp.getValue());
+                    }
+
+                    Property artifactResolveUrlProp = IdPManagementUIUtil.getProperty(fedAuthnConfig.getProperties(),
+                        IdentityApplicationConstants.Authenticator.SAML2SSO.ARTIFACT_RESOLVE_URL);
+                    if (artifactResolveUrlProp !=null) {
```
Comment (thanujalk) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp]: Format issue

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp [hunk 5]
```diff
@@ -560,6 +564,24 @@
...
+                        artifactResolveUrl = artifactResolveUrlProp.getValue();
+                    }
+
+                    Property artifactResolveReqSignedProp = IdPManagementUIUtil.getProperty(fedAuthnConfig.getProperties(),
+                        IdentityApplicationConstants.Authenticator.SAML2SSO.IS_ARTIFACT_RESOLVE_REQ_SIGNED);
+                    if (artifactResolveReqSignedProp !=null) {
```
Comment (thanujalk) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp]: Format issue

File: components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java [hunk 6]
```diff
@@ -1617,6 +1617,32 @@ private static void buildSAMLAuthenticationConfiguration(IdentityProvider fedIdp
         }
         properties.add(property);
 
+        property = new Property();
+        property.setName(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_ARTIFACT_BINDING_ENABLED);
+        if ("on".equals(paramMap
+                .get(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_ARTIFACT_BINDING_ENABLED))) {
+            property.setValue("true");
+        } else {
+            property.setValue("false");
```
Comment (pulasthi7) [inline components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/java/org/wso2/carbon/idp/mgt/ui/util/IdPManagementUIUtil.java]: better to use Boolean.FALSE.toString()

**Comment by IndunilRathnayake** [comment]
Added artifact response signing configuration with https://github.com/wso2/carbon-identity-framework/pull/1559

**Pull request merged**